torchrl/data/tensordict/tensordict.py

```diff
@@ -1837,6 +1844,18 @@ def __init__(
 
         if source is not None:
             for key, value in source.items():
+
```

Comment:
I decided to pass all of the parent arguments to the nested child TensorDict, because we're guaranteed to have them at initialisation. Otherwise, we can leave only initial three

Reply:
Ok for `_run_checks`, `_is_shared`, `_is_memmap`, but `_meta_source` should not be passed